octobot_trading/personal_data/orders/order_util.py

```diff
@@ -537,3 +537,30 @@ async def get_order_size_portfolio_percent(exchange_manager, order_amount, side,
 
 def generate_order_id():
     return str(uuid.uuid4())
+
+
+async def wait_for_order_fill(order, timeout, wait_for_portfolio_update):
+    if order.is_open():
+        if order.state is None:
+            logging.get_logger(LOGGER_NAME).error(
+                f"None state on created order, impossible to wait for fill, order: {order}"
+            )
+        else:
+            try:
+                await order.state.wait_for_next_state(timeout)
+            except asyncio.TimeoutError:
+                logging.get_logger(LOGGER_NAME).error(
+                    f"Timeout while waiting for rebalance marker open order fill, order {order}"
+                )
+            if wait_for_portfolio_update and isinstance(order.state, fill_order_state.FillOrderState):
+                # portfolio is updated in FillOrderState: wait for this state to complete
```

Comment:
It concerns local portfolio update right? Or it fetches the exchange?

Reply:
both, actually when it's about local pf, this wait is not even done as order is in closed state. Here it's mostly to wait for the portfolio refresh that is triggered when an order is filled. otherwise we proceed with a filled order and out of sync portfolio